### PR TITLE
fix(module-tools): update umd global code

### PR DESCRIPTION
**Pull request by targeral**
### <samp>🤖 Generated by Copilot at f9b4105</samp>

Update the umd global code in the `@modern-js/module-tools` package to improve compatibility and add a changeset file to document the feature.

### <samp>🤖 Generated by Copilot at f9b4105</samp>

*  Add a changeset file to describe the patch-level update and new feature for the `@modern-js/module-tools` package ([link](https://github.com/web-infra-dev/modern.js/pull/4825/files?diff=unified&w=0#diff-acc3379c259b0f983ddd342f9a9500ac752094bdbb99540f710713108ce16345R1-R6))
*  Update the `adapterPlugin` function in the `esbuild/adapter.ts` file to improve the compatibility of the umd global code by adding a fallback expression and reordering the variables ([link](https://github.com/web-infra-dev/modern.js/pull/4825/files?diff=unified&w=0#diff-a99e93a85a25f2e6a158276b6dda938ff69960872fbcf2c32e40e74f236f32e7L215-R215))

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/solutions/module-tools/src/builder/esbuild/adapter.ts [hunk 1]
```diff
@@ -212,7 +212,7 @@ export const adapterPlugin = (compiler: ICompiler): Plugin => {
         if (args.namespace === globalNamespace) {
           const value = config.umdGlobals[args.path];
           return {
-            contents: `module.exports = (typeof globalThis !== "undefined" ? globalThis : Function('return this')() || global || self)[${JSON.stringify(
+            contents: `module.exports = (typeof globalThis !== "undefined" ? globalThis : global || self || Function('return this')())[${JSON.stringify(
```
Comment (chenjiahan) [inline packages/solutions/module-tools/src/builder/esbuild/adapter.ts:215]: Can you provide some context for this change?

Comment (targeral) [reply packages/solutions/module-tools/src/builder/esbuild/adapter.ts:215]: You can read about `global` and `self` in caniuse.

**Review by 10Derozan — APPROVED** [review]

File: packages/solutions/module-tools/src/builder/esbuild/adapter.ts (hunk 1 above)
Comment (chenjiahan) [reply packages/solutions/module-tools/src/builder/esbuild/adapter.ts:217]: @targeral Will this change break the code in the browsers?

image

Comment (targeral) [reply packages/solutions/module-tools/src/builder/esbuild/adapter.ts]: OK, I will fix it.

Comment (targeral) [reply packages/solutions/module-tools/src/builder/esbuild/adapter.ts]: You can review again.

**Review by chenjiahan — APPROVED** [review]

File: packages/solutions/module-tools/src/builder/esbuild/adapter.ts (hunk 1 above)
Comment (chenjiahan) [reply packages/solutions/module-tools/src/builder/esbuild/adapter.ts]: LGTM 👍

Comment (targeral) [reply packages/solutions/module-tools/src/builder/esbuild/adapter.ts]: Thx

**Pull request merged**